cylc/flow/platforms.py

```diff
@@ -47,21 +47,26 @@ def platform_from_name(platform_name=None, platforms=None):
         platforms = glbl_cfg().get(['platforms'])
 
     if platform_name is None:
-        platform_data = deepcopy(platforms['localhost'])
-        platform_data['name'] = 'localhost'
-        return platform_data
+        return platforms['localhost']
 
     # The list is reversed to allow user-set platforms (which are loaded
     # later than site set platforms) to be matched first and override site
     # defined platforms.
     for platform_name_re in reversed(list(platforms)):
         if re.fullmatch(platform_name_re, platform_name):
-            platform_data = platforms[platform_name_re]
-            if not platform_data:
-                platform_data = deepcopy(platforms['localhost'])
-                platform_data['remote hosts'] = platform_name
-            else:
-                platform_data = deepcopy(platform_data)
+            # Deepcopy prevents contaminating platforms with data
```

Comment:
Without deepcopy I was getting failures on the test for vld456 returning data from the test for vld123 which had already been loaded. :disappointed: